File: go/0001-two-sum.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, val := range nums {
		diff := target - val
		if mapIndex, ok := m[diff]; ok {
			return []int{mapIndex, i}
		}
		m[val] = i
	}
	return []int{}
}

File: python/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        twoSumHash = {}
        for i,n in enumerate(nums):
            diff = target - n
            if diff in twoSumHash:
                return [twoSumHash[diff], i]
            twoSumHash[n] = i
